fdfgen/__init__.py

```diff
@@ -57,6 +59,9 @@ def handle_data_strings(fdf_data_strings, fields_hidden, fields_readonly):
 
 
 def handle_data_names(fdf_data_names, fields_hidden, fields_readonly):
+    if isinstance(fdf_data_names, dict):
+        fdf_data_names = fdf_dfdf_data_namesata_strings.items()
```

Comment:
Sorry, yes, a terrible copy and paste error (I only tested out the data_strings stuff that I changed, but it made sense to access dicts for data_names too)!